rover_video/src/camera_node.cpp

```diff
@@ -0,0 +1,558 @@
+    {
+        RCLCPP_ERROR(LOGGER, "Failed to create screenshots folder or it already exists.");
+        response_.success = false;
+        response_.status = "Failed to create screenshots folder or it already exists.";
+        // Need to find how to handle
+    }
+    if (this->newRecording(folderPath, captureName, cameraURL))
+    {
+        response_.success = true;
+        response_.status = "Recording started";
+    }
+    else
+    {
+        response_.success = false;
+        response_.status = "Failed to take a video.";  // add reason i.e. recording already started at TIME-GPS-NAME
+    }
+}
+
+/**
+ * @brief Executes the program in order when a recording stops
+ *
+ * @param request_
+ * @param response_
+ */
+void CameraNode::stopRecordingLogic(const rover_msgs::srv::CameraControl::Request& request_,
+                                    rover_msgs::srv::CameraControl::Response& response_)
+{
+    std::string cameraURL = request_.camera_url;
+
+    if (this->stopRecording(cameraURL))
+    {
+        response_.success = true;
+        response_.status = "Recording ended";
+    }
+    else
+    {
+        response_.success = false;
+        response_.status = "No such recordings";
+    }
+}
+
+/**
+ * @brief Get the time the request was made
+ *
+ * @return std::string
+ */
+std::string CameraNode::getCurrentTime(void)
+{
+    std::chrono::time_point<std::chrono::system_clock> now = std::chrono::system_clock::now();  // get system time
+
+    std::time_t now_time = std::chrono::system_clock::to_time_t(now);  // convert to real time
+
+    std::tm tm_now = *std::localtime(&now_time);  // convert to calendar time
+
+    std::stringstream current_time_output;
+    current_time_output << std::put_time(&tm_now, "%FT%T");  // ISO 8601 format
+
+    return current_time_output.str();
+}
+
+/**
+ * @brief Gets the filename necessary to save the file, depending on the file type
+ *
+ * @param capture_name_ Custom name made by the user
+ * @param camURL_ The rtsp url for the camera in use
+ * @param fileType_ Whether it is a screenshot or a video
+ * @return std::string of the complete filename
+ */
+std::string CameraNode::getFileName(const std::string& capture_name_, std::string camURL_, eFileFormatNameTypes fileType_)
+{
+    std::string filename;
+
+    std::string time = this->getCurrentTime();
+    std::string latitude = std::to_string(_last_latitude);
+    std::string longitude = std::to_string(_last_longitude);
+    std::string ID = "UnknownID";
+    auto it = CameraInfo::CameraName.find(camURL_);
+
+    if (it != CameraInfo::CameraName.end())
+    {
+        ID = it->second;
+    }
+
+    switch (fileType_)
+    {
+        case eFileFormatNameTypes::SCREENSHOT:
+            filename = capture_name_.empty()
+                           ? time + "_lat:" + latitude + "_long:" + longitude + "_" + ID + "_screenshot.png"
+                           : time + "_lat:" + latitude + "_long:" + longitude + "_camID:" + ID + "_" + capture_name_;
```

Comment:
Goes with another comment: Please force the extension and consider the "capture_name_" user input as without extension.
```suggestion
                           : time + "_lat:" + latitude + "_long:" + longitude + "_camID:" + ID + "_" + capture_name_ + ".png";
```